spinalcordtoolbox/scripts/sct_register_to_template.py

```diff
@@ -612,8 +607,17 @@ def main(argv: Sequence[str]):
                     f"the labels being outside the ROI of the spinal cord segmentation. Please make sure all labels "
                     f"are within the ROI of the spinal cord segmentation.", type='error'
                 )
-
-            # Compute rigid transformation straight landmarks --> template landmarks
+            
+            # Add labels in the X-Z and Y-Z planes to accommodate for rotations about the Y and X axes
+            printv('\nAdd labels in the X-Z and Y-Z planes to accommodate for rotations about the Y and X axes...', verbose)
+            # -> Pick a dummy label between [1, 127] that doesn't clash with the existing label values.
+            existing_label_vals = {coord.value for coord in img_tmp_label.getNonZeroCoordinates()}
+            positive_int8_vals = set(range(1, 128))
+            dummy_label = max(positive_int8_vals - existing_label_vals)  # Should be 127 in 99.99% of cases
+            add_orthogonal_label(ftmp_label, new_label_value=dummy_label)
+            add_orthogonal_label(ftmp_template_label, new_label_value=dummy_label)
```

Comment:
This newly added code is duplicated from the `ref == 'subject'` case, right?

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/634433f9572e451eaae9a7d8bc94d9d52ff17461/spinalcordtoolbox/scripts/sct_register_to_template.py#L770-L776

That makes sense, because now both cases compute an affine transformation (incl. rotation) for the pre-registration step. 

I'm wondering if maybe it's worth creating a function for this logic, then calling the function in both places (to avoid duplication)?

Reply:
> I'm wondering if maybe it's worth creating a function for this logic, then calling the function in both places (to avoid duplication)?

yup, i thought about it, but unfortunately no time to do it-- could be for subsequent PR maybe?